src/libproxychains.c

```diff
@@ -682,7 +686,13 @@ int pc_getnameinfo(const struct sockaddr *sa, socklen_t salen,
 	return 0;
 }
 
+#ifdef IS_HAIKU
+struct hostent *gethostbyaddr(const char *address, socklen_t length, int type) {
```

Comment:
i'd rather not have this ugly condition here, thus i filed https://github.com/haiku/haiku/pull/20

Reply:
i have a better patch for haiku available here https://0x0.st/iYLN.patch
in case you have an account for their trac, would you mind posting it there?